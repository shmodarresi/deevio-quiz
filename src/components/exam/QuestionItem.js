import React, { useState } from 'react';


const QuestionItem = ({item,selectAnswer,score}) => {

    const [isComplete , setIsComplete] = useState('');
    const [isCorrect , setIsCorrect] = useState('');


    //create array of correct answer and incorrect answer
    const answers = () => {
        var array = [...item.incorrect_answers , item.correct_answer];
        return array;
    }
    
    //calculate score and send it to Quest list component
    const selectFn = (answer , e) => {

        if(answer === item.correct_answer){
            e.target.className = 'isCorrect';
            score(1);
        }else{
            e.target.className = 'isWrong';
            setIsCorrect('isCorrect');
            score(0);
        }

        setIsComplete(true);
        selectAnswer(answer);
    }

    return (
        <li className='card p-3'>
            <div dangerouslySetInnerHTML={{__html: item.question}}></div>
                {/* class disabled = doesn't let to answer the question more than one time*/}
                <ul className={isComplete?'disabled' : ''}> 
                    {answers().map(a => {
                        return(
                        <li  className={item.correct_answer === a? isCorrect :''}
                            key={answers().indexOf(a)}
                            onClick={(e) => selectFn(a,e)} >
                            <span className="rounded-circle mr-1"></span>
                            <span dangerouslySetInnerHTML={{__html: a}}></span>
                        </li>
                        )
                    })}
                </ul>
        </li>
    );
};

export default QuestionItem;