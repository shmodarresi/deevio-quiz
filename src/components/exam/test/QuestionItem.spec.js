import React from 'react';
import QuestionItem from '../QuestionItem';
import renderer from 'react-test-renderer';

let q={question:'test item',correct_answer:'true',incorrect_answers:['false']};

describe("the question item test",() =>{
    test("renders as expeted",() => {
        const question = renderer.create(<QuestionItem item={q} />);
        expect(question.toJSON()).toMatchSnapshot();

    })
})