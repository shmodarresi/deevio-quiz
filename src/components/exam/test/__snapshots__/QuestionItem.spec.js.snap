// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the question item test renders as expeted 1`] = `
<li
  className="card p-3"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "test item",
      }
    }
  />
  <ul
    className=""
  >
    <li
      className=""
      onClick={[Function]}
    >
      <span
        className="rounded-circle mr-1"
      />
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "false",
          }
        }
      />
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      <span
        className="rounded-circle mr-1"
      />
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "true",
          }
        }
      />
    </li>
  </ul>
</li>
`;
